cirq-core/cirq/sim/act_on_args.py

```diff
@@ -113,14 +115,31 @@ def _perform_measurement(self, qubits: Sequence['cirq.Qid']) -> List[int]:
         """Child classes that perform measurements should implement this with
         the implementation."""
 
-    def copy(self: TSelf) -> TSelf:
-        """Creates a copy of the object."""
+    def copy(self: TSelf, deep_copy_buffers: bool = True) -> TSelf:
+        """Creates a copy of the object.
+
+        Args:
+            deep_copy_buffers: If True, buffers will also be copied.
```

Comment:
Let's just be more thorough -- If True, buffers will also be deep-copied. Otherwise the copy will share a reference to the original object's buffers.

Reply:
I finally understand the word *reference* and *reuse* now! Please check my last [commit](https://github.com/quantumlib/Cirq/pull/4789/commits/62addd16b88803634d5e65f335f5f9f54a799b61). I think it is what you intended to see. Sorry that I missed your point all this long time!